packages/maskbook/src/components/InjectedComponents/PostDialog.tsx

```diff
@@ -456,13 +456,15 @@ export function CharLimitIndicator({ value, max, ...props }: CircularProgressPro
 }
 
 function PluginRenderer() {
-    const chainIdValid = useChainIdValid()
+    const chainId = useChainId()
     const result = useActivatedPluginsSNSAdaptor().map((plugin) =>
         Result.wrap(() => {
             const entry = plugin.CompositionDialogEntry
             const unstable = plugin.enableRequirement.target !== 'stable'
-            const requireChainValid = Boolean(plugin.enableRequirement.web3?.compositionEntryRequiresChainIDValid)
-            if (!entry || (!chainIdValid && requireChainValid)) return null
+            const chainIdValid =
+                !Boolean(plugin.enableRequirement.web3?.compositionEntryRequiredChains) ||
+                Boolean(plugin.enableRequirement.web3?.compositionEntryRequiredChains?.includes(chainId))
+            if (!entry || !chainIdValid) return null
```

Comment:
Does this change mean that if the user chose an unmatched chain, the plugin would not render?

Reply:
Never mind. It's OK to not render the composition entry.

Reply:
Another choice is to render the composition entry but in a disabled chip with a tip "this plugin is not available in the current network". That might be less confusing.